session/tidb.go

```diff
@@ -192,6 +192,10 @@ func runStmt(ctx context.Context, sctx sessionctx.Context, s sqlexec.Statement)
 	var err error
 	var rs sqlexec.RecordSet
 	se := sctx.(*session)
+	err = se.isTxnAborted(s)
+	if err != nil {
+		return nil, errors.Trace(err)
```

Comment:
if the problem only ocur in retry, we can don't change this and `changeInvalidToPending`?